test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -451,122 +446,51 @@ protected bool CheckOutputIsSuffix(string basePath, string outPath, int skip = 0
             }
         }
 
-        protected bool CheckEqualityFromPathsCore(string relPath, string basePath, string outPath, int skip = 0,
+        protected void CheckEqualityFromPathsCore(string relPath, string basePath, string outPath, int skip = 0,
             int digitsOfPrecision = DigitsOfPrecision, NumberParseOption parseOption = NumberParseOption.Default)
         {
             Contracts.Assert(skip >= 0);
 
-            using (StreamReader baseline = OpenReader(basePath))
-            using (StreamReader result = OpenReader(outPath))
-            {
-                int count = 0;
-                if (skip > 0)
-                {
-                    string line2;
-                    do
-                    {
-                        line2 = result.ReadLine();
-                        if (line2 == null)
-                        {
-                            Fail("Output is shorter than the skip value of {0}!", skip);
-                            return false;
-                        }
-                        count++;
-                    } while (count <= skip);
-
-                    string line1;
-                    do
-                    {
-                        line1 = baseline.ReadLine();
-                        if (line1 == null)
-                        {
-                            Fail("Couldn't match output file line to a line in the baseline!");
-                            return false;
-                        }
-                    } while (line1 != line2);
-                }
-
-                for (; ; )
-                {
-                    // read lines while we can
-                    string line1 = baseline.ReadLine();
-                    string line2 = result.ReadLine();
-
-                    if (line1 == null && line2 == null)
-                    {
-                        Log("Output matches baseline: '{0}'", relPath);
-                        return true;
-                    }
-
-                    count++;
-                    var inRange = GetNumbersFromFile(ref line1, ref line2, digitsOfPrecision, parseOption);
-
-                    if (!inRange || line1 != line2)
-                    {
-                        if (line1 == null || line2 == null)
-                            Fail("Output and baseline different lengths: '{0}'", relPath);
-                        else
-                            Fail(_allowMismatch, "Output and baseline mismatch at line {1}, expected '{2}' but got '{3}' : '{0}'", relPath, count, line1, line2);
-                        return false;
-                    }
-                }
-            }
+            var baseline = ReadFileWithNumericPrecision(basePath, skip, digitsOfPrecision, parseOption);
+            var result = ReadFileWithNumericPrecision(outPath, skip, digitsOfPrecision, parseOption);
+            AssertEx.EqualOrDiff(baseline, result, $"Actual and expected values differ for '{relPath}'. Expected shown in baseline of diff:");
```

Comment:
📝 This is the motivating change of this pull request. Rather than compare the numbers in the file one at a time, we compare the full normalized contents at once.

Reply:
and what difference does that make? is this just for better debugging to see the diff>

---
In reply to: [356927868](https://github.com/dotnet/machinelearning/pull/4565#discussion_r356927868) [](ancestors = 356927868)